File: ProiectColectiv/ProiectColectiv.Application/Models/SpecialisationModel.cs
using ProiectColectiv.Domain.Enums;

namespace ProiectColectiv.Application.Models.Specializations
{
    public class SpecializationModel
    {
        public string Name { get; set; }
        public int NoOfYears { get; set; }
        public MajorEnum Major { get; set; }

        public List<SubjectModel> Subjects { get; set; } = new();
    }
}

File: proiect-colectiv-frontend/src/app/models/specialisations/SpecialisationModel.ts
import { BaseModel } from '../base/BaseModel';
import { MajorEnum } from '../enums/MajorEnums';
import { SubjectModel } from '../subjects/SubjectModel';

export class SpecialisationModel extends BaseModel {
  name: string;
  noOfYears: number;
  major: MajorEnum;
  subjects: SubjectModel[];

  constructor(
    id: string,
    name: string,
    noOfYears: number,
    major: MajorEnum,
    subjects: SubjectModel[]
  ) {
    super(id);
    this.name = name;
    this.noOfYears = noOfYears;
    this.major = major;
    this.subjects = subjects;
  }
}
